crates/forge/src/main.rs

```diff
@@ -118,9 +139,18 @@ fn main_execution() -> Result<bool> {
     if args.clean_cache {
         clean_cache(&workspace_root).context("Failed to clean snforge cache")?;
     }
-    let cores_approx = available_parallelism()?.get();
+
+    let cores = if let Some(cores) = args.cores {
+        cores
+    } else if let Ok(available_cores) = available_parallelism() {
+        available_cores.get()
+    } else {
+        eprintln!("Failed to get the number of available cores, defaulting to 1");
+        1
+    };
```

Comment:
I'm not sure if we should be defaulting to 1 at all. Previous approach where we bubbled an error seemed reasonable.

Reply:
I wonder in which cases `available_parallelism` can return an error at all. I guess there is some benefit of having a default.